### PR TITLE
planner, executor: fix rename `sm_join` to `merge_join`

### LINKED ISSUE #15200: rename `sm_join` to `merge_join`
## Feature Request

**Is your feature request related to a problem? Please describe:**

these hints are used to control the join algorithms:

* Hash Join: hash_join()
* Merge Join: sm_join()
* Index Hash Join: inl_hash_join()
* Index Merge join: inl_merge_join()

It's better to change `sm_join()` to `merge_join()` for these reasons:
* What is `sm` short for? It makes user confused.
* `merge_join()`/`inl_merge_join()` and `hash_join()`/`inl_hash_join()`, keeping the same naming pattern makes people much easier to remember these 4 hints.

**Describe the feature you'd like:**

As the title said, change `sm_join` to `merge_join`, merge it to release 4.0 before it's GA.

**Describe alternatives you've considered:**

No

**Teachability, Documentation, Adoption, Migration Strategy:**

No

### PR DESCRIPTION
**Pull request by lawyerphx**
### What problem does this PR solve? 
Fix #15200 

### What is changed and how it works?
- fix rename `sm_join` to `merge_join`
- redirect parser

### Check List 

Tests 

 - Unit test
 - Integration test
 - Manual test (add detailed scripts or steps below)
 - No code

Code changes

 - Has exported variable/fields change
 - Has persistent data change

Side effects

Related changes

 - Need to update the documentation
 - Need to update the `tidb-ansible` repository

Release note

 - Waiting for [parser PR](https://github.com/pingcap/parser/pull/775) to be fixed, need to update the version code of pingcap/parser in go.mod

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -76,3 +76,5 @@ require (
 )
 
 go 1.13
+
+replace github.com/pingcap/parser => github.com/lawyerphx/parser v0.0.0-20200316023126-7af13c9e0b85
```
Comment (qw4990) [inline go.mod]: Please remove this.

Comment (lawyerphx) [reply go.mod]: OK， waiting for this PR to be merged (#775)[https://github.com/pingcap/parser/pull/775]

**Pull request merged**